# eBay Kleinanzeigen
This a report that analyze the eBay Kleinanzeigen data set of the German ebay Website.

In [1]:
import pandas as pd
import numpy as np
autos = pd.read_csv('autos.csv',encoding='Latin-1')

In [2]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [3]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

Our dataset here has 50000 rows and 20 columns. There are missing values on vehicleType, gearbox, model, fuelType and notRepairedDamage columns. There are five columns with integer type while the rest are in object type.

In [4]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

Thre are three column that have data of date but they are set as object type which we need to change to datetime data type. also odometer column supposed to be numeric column so we can make calculation on it but it need first to extract numeric data from it and convert its type into numeric. Also Price column is the same as odometer column. The postal Code column is set as numeric column which is not supposed to be because postal codes are calssification features and not supposed to make calculations on them so we need to change this column into a string column.

In [5]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [6]:
new_columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_pictures', 'postal_code',
       'last_seen']
autos.columns = new_columns

In [7]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_pictures  \
0     peugeot              nein  2016-03-26 00:00:00            0   
1         bmw              nein  2016-04-04 00:00:00            0   
2  volkswagen              nein  2016-03-26 00:00:00            0   
3       smart              nein  2016-03-12 00:00:00            0   
4        ford              nein  2016-04-01 00:00:00            0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

In the previous piece of code I've changed the column names from camel case into snake case which will make it easier to handle the dataset with.

In [8]:
autos.describe()

registration_year      power_ps  registration_month  nr_pictures  \
count       50000.000000  50000.000000        50000.000000      50000.0   
mean         2005.073280    116.355920            5.723360          0.0   
std           105.712813    209.216627            3.711984          0.0   
min          1000.000000      0.000000            0.000000          0.0   
25%          1999.000000     70.000000            3.000000          0.0   
50%          2003.000000    105.000000            6.000000          0.0   
75%          2008.000000    150.000000            9.000000          0.0   
max          9999.000000  17700.000000           12.000000          0.0   

        postal_code  
count  50000.000000  
mean   50813.627300  
std    25779.747957  
min     1067.000000  
25%    30451.000000  
50%    49577.000000  
75%    71540.000000  
max    99998.000000

Calculating the statistical values for month , year and postal code have no sense so we need to exclude these columns from our statistical analysis. Also there's a column that all values are zeros called nr_pictures of course wedon't need to calculate anything for this column

There are columns we need to include in our statistical calculations are not included here which means that they are stored as text so we need to turn them into numeric colunms

In [9]:
autos['price']= autos['price'].str.replace('$','')
autos['price']= autos['price'].str.replace(',','')
autos['odometer']= autos['odometer'].str.replace('km','')
autos['odometer']= autos['odometer'].str.replace(',','')

In [10]:
autos.rename({'odometer':'odometer_km'})
autos['price']=pd.to_numeric(autos['price'])
autos['odometer']=pd.to_numeric(autos['odometer'])

In [11]:
autos['price'].unique().shape

(2357,)

In [12]:
desc = autos['price'].describe()

In [13]:
autos['price'].value_counts().head()

0       1421
500      781
1500     734
2500     643
1000     639
Name: price, dtype: int64

In [14]:
autos['price'].value_counts().sort_index(ascending=False).head(10)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
Name: price, dtype: int64

In [15]:
autos = autos[autos['price'].between(desc[4],desc[6])]

In [16]:
autos['price'].describe()

count    25369.000000
mean      3287.426229
std       1705.045371
min       1100.000000
25%       1800.000000
50%       2900.000000
75%       4500.000000
max       7200.000000
Name: price, dtype: float64

In [17]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 25369 entries, 0 to 49999
Data columns (total 20 columns):
date_crawled          25369 non-null object
name                  25369 non-null object
seller                25369 non-null object
offer_type            25369 non-null object
price                 25369 non-null int64
abtest                25369 non-null object
vehicle_type          23222 non-null object
registration_year     25369 non-null int64
gearbox               24241 non-null object
power_ps              25369 non-null int64
model                 24265 non-null object
odometer              25369 non-null int64
registration_month    25369 non-null int64
fuel_type             23537 non-null object
brand                 25369 non-null object
unrepaired_damage     20827 non-null object
ad_created            25369 non-null object
nr_pictures           25369 non-null int64
postal_code           25369 non-null int64
last_seen             25369 non-null object
dtypes: int64(7), 

In [18]:
autos['postal_code'] = autos['postal_code'].astype(str)

/dataquest/system/env/python3/lib/python3.4/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


In [19]:
autos[['date_crawled','ad_created','last_seen']].head()

date_crawled           ad_created            last_seen
0   2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
3   2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4   2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50
7   2016-03-16 18:55:19  2016-03-16 00:00:00  2016-04-07 03:17:32
12  2016-03-31 19:48:22  2016-03-31 00:00:00  2016-04-06 14:17:52

In [20]:
print(autos['date_crawled'].str[:10])

0        2016-03-26
3        2016-03-12
4        2016-04-01
7        2016-03-16
12       2016-03-31
13       2016-03-23
14       2016-03-23
17       2016-03-29
19       2016-03-17
20       2016-03-05
28       2016-03-19
29       2016-04-02
31       2016-03-14
33       2016-03-15
34       2016-04-05
35       2016-03-29
38       2016-03-21
39       2016-03-11
40       2016-03-07
41       2016-03-10
47       2016-03-28
50       2016-03-09
51       2016-03-22
52       2016-03-25
54       2016-03-15
57       2016-03-12
58       2016-03-30
60       2016-03-23
62       2016-03-12
66       2016-03-14
            ...    
49938    2016-03-28
49940    2016-03-11
49945    2016-04-04
49950    2016-03-15
49951    2016-03-31
49952    2016-03-21
49954    2016-03-17
49955    2016-03-30
49956    2016-03-28
49957    2016-03-10
49958    2016-03-08
49959    2016-03-28
49961    2016-03-14
49962    2016-03-14
49965    2016-03-11
49966    2016-04-02
49967    2016-03-12
49968    2016-04-01
49969    2016-03-17


In [21]:
autos['date_crawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=False)

2016-04-07    0.001458
2016-04-06    0.003784
2016-04-05    0.013047
2016-04-04    0.037211
2016-04-03    0.038669
2016-04-02    0.035319
2016-04-01    0.032481
2016-03-31    0.031771
2016-03-30    0.033190
2016-03-29    0.034018
2016-03-28    0.035161
2016-03-27    0.031535
2016-03-26    0.033387
2016-03-25    0.029564
2016-03-24    0.028263
2016-03-23    0.031692
2016-03-22    0.033466
2016-03-21    0.037447
2016-03-20    0.037684
2016-03-19    0.034097
2016-03-18    0.012732
2016-03-17    0.030431
2016-03-16    0.029524
2016-03-15    0.033466
2016-03-14    0.037211
2016-03-13    0.014979
2016-03-12    0.038039
2016-03-11    0.034254
2016-03-10    0.033860
2016-03-09    0.033111
2016-03-08    0.033584
2016-03-07    0.035792
2016-03-06    0.014033
2016-03-05    0.025740
Name: date_crawled, dtype: float64

In [22]:
autos['date_crawled'] = pd.to_datetime(autos['date_crawled'])
autos['ad_created'] = pd.to_datetime(autos['ad_created'])
autos['last_seen'] = pd.to_datetime(autos['last_seen'])

/dataquest/system/env/python3/lib/python3.4/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
/dataquest/system/env/python3/lib/python3.4/site-packages/ipykernel/__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
/dataquest/system/env/python3/lib/python3.4/site-packages/ipykernel/__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value

In [23]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 25369 entries, 0 to 49999
Data columns (total 20 columns):
date_crawled          25369 non-null datetime64[ns]
name                  25369 non-null object
seller                25369 non-null object
offer_type            25369 non-null object
price                 25369 non-null int64
abtest                25369 non-null object
vehicle_type          23222 non-null object
registration_year     25369 non-null int64
gearbox               24241 non-null object
power_ps              25369 non-null int64
model                 24265 non-null object
odometer              25369 non-null int64
registration_month    25369 non-null int64
fuel_type             23537 non-null object
brand                 25369 non-null object
unrepaired_damage     20827 non-null object
ad_created            25369 non-null datetime64[ns]
nr_pictures           25369 non-null int64
postal_code           25369 non-null object
last_seen             25369 non-null datetime

In [24]:
autos['ad_created'].describe()

count                   25369
unique                     60
top       2016-04-03 00:00:00
freq                      988
first     2015-08-10 00:00:00
last      2016-04-07 00:00:00
Name: ad_created, dtype: object

In [25]:
autos['date_crawled'].describe()

count                   25369
unique                  24923
top       2016-04-02 11:37:04
freq                        3
first     2016-03-05 14:06:30
last      2016-04-07 14:36:44
Name: date_crawled, dtype: object

In [26]:
autos['last_seen'].describe()

count                   25369
unique                  22507
top       2016-04-05 20:47:37
freq                        5
first     2016-03-05 14:49:34
last      2016-04-07 14:58:50
Name: last_seen, dtype: object

In [27]:
autos['registration_year'].value_counts().head()

2003    2205
2004    2179
2005    2155
2002    1964
2001    1888
Name: registration_year, dtype: int64

In [28]:
autos['registration_month'].value_counts().head()

3    2661
6    2322
5    2174
4    2135
7    1983
Name: registration_month, dtype: int64

From the previous analysis on dates columns we can see that 2016-04-03 was the most date the ad created on also we can see that 2003 was the most year that cars got regestered and the most month cars get regestered was March.

In [31]:
autos['registration_year'].sort_values()

22316    1000
2221     1934
2573     1934
21421    1937
23804    1937
23372    1952
35453    1953
23483    1956
19914    1956
20105    1958
10358    1958
20471    1959
7094     1960
4942     1960
23457    1960
35092    1960
893      1960
49270    1960
38449    1961
44038    1962
14265    1962
25114    1962
19024    1963
33835    1963
29664    1964
12647    1964
33897    1964
44469    1965
47390    1965
18551    1965
         ... 
17761    2018
36190    2018
24834    2018
9950     2018
22055    2018
6360     2018
36227    2018
11722    2018
42396    2018
8792     2018
18638    2018
6376     2018
23998    2018
36258    2018
7664     2018
14619    2018
17314    2018
41042    2018
12185    2018
41045    2018
20620    2018
18662    2018
11111    2018
7201     2018
2089     2018
49185    2019
27578    2800
4549     4100
49153    5000
25003    8888
Name: registration_year, Length: 25369, dtype: int64

From what we see in the previous cell that there's unrealistic values for the registeration year colum like 1000 whish is 1019 years ago and (2800,4100,5000 and 8888) which are years on the far (very far) future. So we need to remove data fall in these unrealistic values.

In [32]:
autos = autos.loc[(autos['registration_year']<2019)&(autos['registration_year']>1900)]

In [33]:
autos['registration_year'].value_counts(normalize=True)

2003    0.086938
2004    0.085913
2005    0.084966
2002    0.077436
2001    0.074439
2000    0.071443
2006    0.070339
1999    0.062414
2007    0.049955
1998    0.044908
2008    0.038205
2009    0.033513
2017    0.031108
1997    0.027520
2016    0.019122
1996    0.018334
2010    0.015337
1995    0.015258
2018    0.012814
2011    0.009226
1994    0.009187
1993    0.007097
1991    0.006782
2012    0.006624
1992    0.006545
1990    0.006151
1989    0.003943
1988    0.003391
2013    0.002129
1985    0.002050
          ...   
2014    0.000907
1978    0.000749
1981    0.000591
1970    0.000591
1973    0.000434
1977    0.000434
1972    0.000394
1979    0.000394
2015    0.000315
1971    0.000276
1976    0.000276
1969    0.000276
1975    0.000237
1960    0.000237
1966    0.000237
1965    0.000237
1968    0.000197
1967    0.000197
1974    0.000197
1962    0.000118
1964    0.000118
1958    0.000079
1937    0.000079
1934    0.000079
1963    0.000079
1956    0.000079
1953    0.000039
1961    0.0000

In [34]:
autos['registration_year'].max()

2018

In [35]:
autos['registration_year'].min()

1934

In [40]:
brands = autos['brand'].value_counts().reset_index()

In [41]:
brands

index  brand
0       volkswagen   5410
1             opel   2860
2              bmw   2515
3    mercedes_benz   2468
4             audi   1824
5             ford   1716
6          renault   1246
7          peugeot    904
8             fiat    685
9            smart    589
10            seat    507
11           skoda    460
12          toyota    433
13         citroen    424
14           mazda    421
15          nissan    345
16         hyundai    279
17           volvo    237
18           honda    216
19      mitsubishi    213
20             kia    204
21      alfa_romeo    180
22          suzuki    172
23       chevrolet    167
24  sonstige_autos    161
25            mini    140
26        chrysler     91
27           dacia     82
28        daihatsu     61
29          subaru     48
30            jeep     46
31            saab     44
32      land_rover     36
33          jaguar     34
34          daewoo     34
35         trabant     32
36            lada     25
37           rover     23
38          lancia     18
39         porsche     13

In [43]:
prec=[]
for i in range(len(brands)):
    prec.append((brands.iloc[i]['brand']/len(autos['brand']))*100)
brands['precentage']=prec

In [44]:
brands

index  brand  precentage
0       volkswagen   5410   21.330284
1             opel   2860   11.276269
2              bmw   2515    9.916019
3    mercedes_benz   2468    9.730710
4             audi   1824    7.191578
5             ford   1716    6.765761
6          renault   1246    4.912668
7          peugeot    904    3.564247
8             fiat    685    2.700785
9            smart    589    2.322280
10            seat    507    1.998975
11           skoda    460    1.813666
12          toyota    433    1.707211
13         citroen    424    1.671727
14           mazda    421    1.659898
15          nissan    345    1.360249
16         hyundai    279    1.100028
17           volvo    237    0.934432
18           honda    216    0.851634
19      mitsubishi    213    0.839806
20             kia    204    0.804321
21      alfa_romeo    180    0.709695
22          suzuki    172    0.678153
23       chevrolet    167    0.658439
24  sonstige_autos    161    0.634783
25            mini    140    0.551985
26        chrysler     91    0.358790
27           dacia     82    0.323306
28        daihatsu     61    0.240508
29          subaru     48    0.189252
30            jeep     46    0.181367
31            saab     44    0.173481
32      land_rover     36    0.141939
33          jaguar     34    0.134054
34          daewoo     34    0.134054
35         trabant     32    0.126168
36            lada     25    0.098569
37           rover     23    0.090683
38          lancia     18    0.070970
39         porsche     13    0.051256

In [45]:
top_brands = brands.loc[brands['precentage']>5]
top_brands

index  brand  precentage
0     volkswagen   5410   21.330284
1           opel   2860   11.276269
2            bmw   2515    9.916019
3  mercedes_benz   2468    9.730710
4           audi   1824    7.191578
5           ford   1716    6.765761

In [ ]:
autos.column

In [60]:
groups = autos.groupby('brand')
brands_avg_price={}
for name,group in groups:
    if name in top_brands['index'].values:
        brands_avg_price[name]=(group['price'].mean(),group['mileage'].mean())


KeyError: 'mileage'

In [59]:
print(brands_avg_price)

{'ford': 3045.4726107226106, 'opel': 2974.731118881119, 'volkswagen': 3244.609426987061, 'mercedes_benz': 3548.0713128038897, 'bmw': 3593.724055666004, 'audi': 3501.691337719298}


From the top brands we have here we can see that the average price between 2974 to 3593 dollars and that the cheapest brand of them all is OPEL and the most expensive brand of them all is BMW